IDE/ARDUINO/PREPENDED_README.md

```diff
@@ -0,0 +1,196 @@
+feedback reports dramatically better performance when using wolfSSL over
+OpenSSL.
+
+wolfSSL is powered by the wolfCrypt cryptography library. Two versions of
+wolfCrypt have been FIPS 140-2 validated (Certificate #2425 and
+certificate #3389). FIPS 140-3 validation is in progress. For additional
+information, visit the [wolfCrypt FIPS FAQ](https://www.wolfssl.com/license/fips/)
+or contact fips@wolfssl.com.
+
+## Why Choose wolfSSL?
+
+There are many reasons to choose wolfSSL as your embedded, desktop, mobile, or
+enterprise SSL/TLS solution. Some of the top reasons include size (typical
+footprint sizes range from 20-100 kB), support for the newest standards
+(SSL 3.0, TLS 1.0, TLS 1.1, TLS 1.2, TLS 1.3, DTLS 1.0, DTLS 1.2, and DTLS 1.3),
+current and progressive cipher support (including stream ciphers), multi-platform,
+royalty free, and an OpenSSL compatibility API to ease porting into existing
+applications which have previously used the OpenSSL package. For a complete
+feature list, see [Chapter 4](https://www.wolfssl.com/docs/wolfssl-manual/ch4/)
+of the wolfSSL manual.
+
+## Notes, Please Read
+
+### Note 1
+wolfSSL as of 3.6.6 no longer enables SSLv3 by default.  wolfSSL also no longer
+supports static key cipher suites with PSK, RSA, or ECDH. This means if you
+plan to use TLS cipher suites you must enable DH (DH is on by default), or
+enable ECC (ECC is on by default), or you must enable static key cipher suites
+with one or more of the following defines:
+
+```
+WOLFSSL_STATIC_DH
+WOLFSSL_STATIC_RSA
+WOLFSSL_STATIC_PSK
+```
+Though static key cipher suites are deprecated and will be removed from future
+versions of TLS.  They also lower your security by removing PFS.
+
+When compiling `ssl.c`, wolfSSL will now issue a compiler error if no cipher
+suites are available. You can remove this error by defining
+`WOLFSSL_ALLOW_NO_SUITES` in the event that you desire that, i.e., you're
+not using TLS cipher suites.
+
+### Note 2
+wolfSSL takes a different approach to certificate verification than OpenSSL
+does. The default policy for the client is to verify the server, this means
+that if you don't load CAs to verify the server you'll get a connect error,
+no signer error to confirm failure (-188).
+
+If you want to mimic OpenSSL behavior of having `SSL_connect` succeed even if
+verifying the server fails and reducing security you can do this by calling:
+
+```c
+wolfSSL_CTX_set_verify(ctx, WOLFSSL_VERIFY_NONE, NULL);
+```
+
+before calling `wolfSSL_new();`. Though it's not recommended.
+
+### Note 3
+The enum values SHA, SHA256, SHA384, SHA512 are no longer available when
+wolfSSL is built with `--enable-opensslextra` (`OPENSSL_EXTRA`) or with the
+macro `NO_OLD_SHA_NAMES`. These names get mapped to the OpenSSL API for a
+single call hash function. Instead the name `WC_SHA`, `WC_SHA256`, `WC_SHA384` and
+`WC_SHA512` should be used for the enum name.
+
+
+# wolfSSL Release 5.6.6 (Dec 19, 2023)
```

Comment:
Would prefer a link to the ChangeLog.md and not duplicating.

Reply:
This is different from the change log.

My intention for the [PREPENDED_README.md](https://github.com/gojimmypi/wolfssl/blob/PR-Arduino-Update/IDE/ARDUINO/PREPENDED_README.md) is to add some [prepended text](https://github.com/gojimmypi/wolfssl/blob/PR-Arduino-Update/IDE/ARDUINO/Arduino_README_prepend.md) to the beginning of the library `README.md` published in the [Official wolfSSL Arduino Library](https://github.com/wolfSSL/Arduino-wolfSSL/pull/1) to indicate the Arduino-specific nature.

This prepended text might have Arduino-specific release information, outside of the normal wolfSSL versions, that are needed to be tagged as noted in the [Arduino Library Specifications](https://arduino.github.io/arduino-cli/0.35/library-specification/) and noted in the [requirements](https://github.com/arduino/library-registry/blob/main/FAQ.md#what-are-the-requirements-for-a-library-to-be-added-to-library-manager):

> The library repository must have a [Git tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) (or [release](https://docs.github.com/repositories/releasing-projects-on-github/managing-releases-in-a-repository)) and must have been compliant with all the above requirements at the time of that tag.

There is however, an argument that the `PREPENDED_README.md` file is only a temporary file for the published wolfSSL Arduino [README.md](https://github.com/gojimmypi/Arduino-wolfSSL/blob/PR-Initial-Arduino-Library/README.md) and could be excluded here?